engine/crates/wasi-component-loader/src/context.rs

```diff
@@ -19,22 +24,24 @@ use crate::{
 pub struct ChannelLogSender {
     sender: crossbeam::channel::Sender<AccessLogMessage>,
     lossy_log: bool,
+    log_queue_length_counter: UpDownCounter<i64>,
 }
 
 impl ChannelLogSender {
     /// Sends the given access log message to the access log.
     pub fn send(&self, data: AccessLogMessage) -> Result<(), LogError> {
         if self.lossy_log {
-            if let Err(e) = self.sender.try_send(data) {
-                match e {
-                    TrySendError::Full(AccessLogMessage::Data(data)) => return Err(LogError::ChannelFull(data)),
-                    _ => return Err(LogError::ChannelClosed),
-                }
+            match self.sender.try_send(data) {
+                Ok(_) => (),
+                Err(TrySendError::Full(AccessLogMessage::Data(data))) => return Err(LogError::ChannelFull(data)),
+                Err(_) => return Err(LogError::ChannelClosed),
             }
         } else if self.sender.send(data).is_err() {
             return Err(LogError::ChannelClosed);
         }
 
+        self.log_queue_length_counter.add(1, &[]);
```

Comment:
You don't seem to be decreasing it anywhere.

Reply:
I'm not even sure if it works as you access it by name in the federated server. IMO It would be better to include the counter inside a `ChannelLogReceiver` rather than relying on globals

Reply:
There is a test for this, it works. I'll let you hunt where the decreasing happens :)